integtest/src/main/java/org/corfudb/universe/cluster/docker/CorfuServerDockerized.java

```diff
@@ -0,0 +1,324 @@
+        ContainerConfig containerConfig = buildContainerConfig();
+
+        String id;
+        try {
+            ContainerCreation creation = docker.createContainer(containerConfig, params.getGenericName());
+            id = creation.id();
+
+            addShutdownHook(id);
+
+            docker.connectToNetwork(id, docker.inspectNetwork(clusterParams.getNetworkName()).id());
+
+            docker.startContainer(id);
+        } catch (InterruptedException | DockerException e) {
+            Thread.currentThread().interrupt();
+            throw new NodeException("Can't start a container", e);
+        }
+
+        return id;
+    }
+
+    private ContainerConfig buildContainerConfig() throws NodeException {
+        // Bind ports
+        String[] ports = {String.valueOf(params.getPort())};
+        Map<String, List<PortBinding>> portBindings = new HashMap<>();
+        for (String port : ports) {
+            List<PortBinding> hostPorts = new ArrayList<>();
+            hostPorts.add(PortBinding.of("0.0.0.0", port));
+            portBindings.put(port, hostPorts);
+        }
+
+        HostConfig hostConfig = HostConfig.builder()
+                .portBindings(portBindings)
+                .autoRemove(true)
+                .build();
+
+        // Compose command line for starting Corfu
+        String cmdLine = String.format(
+                "java -cp *.jar org.corfudb.infrastructure.CorfuServer %s",
+                getCommandLineParams()
+        );
+
+        try {
+            return ContainerConfig.builder()
+                    .hostConfig(hostConfig)
+                    .image("corfu-server:" + getAppVersion())
+                    .hostname(params.getGenericName())
+                    .exposedPorts(ports)
+                    .cmd("sh", "-c", cmdLine)
+                    .build();
+        } catch (XmlPullParserException | IOException e) {
+            throw new NodeException("Can't parse application version", e);
+        }
+    }
+
+    private void addShutdownHook(String id) {
+        // Just in case if a test failed and didn't kill the container
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                kill();
+            } catch (Exception e) {
+                log.debug("Corfu server shutdown hook. Can't kill container: {}", params.getGenericName());
+            }
+        }));
+    }
+
+    private String getAppVersion() throws IOException, XmlPullParserException {
+        MavenXpp3Reader reader = new MavenXpp3Reader();
+        Model model = reader.read(new FileReader("pom.xml"));
+        return model.getParent().getVersion();
+    }
+
+    /**
+     * This method create a command line string for starting Corfu server
+     *
+     * @return command line parameters
+     */
+    private String getCommandLineParams() {
+        StringBuilder cmd = new StringBuilder();
+        cmd.append("-a ").append("0.0.0.0");
+
+        switch (params.getPersistence()) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)